astropy/units/format/vounit.py

```diff
@@ -98,23 +98,30 @@ def parse(cls, s, debug=False):
         return result
 
     @classmethod
-    def _parse_unit(cls, unit, detailed_exception=True):
-        if unit not in cls._units:
-            if cls._explicit_custom_unit_regex.match(unit):
-                return cls._def_custom_unit(unit)
+    def _get_unit(cls, t):
+        try:
+            super()._get_unit(t)
+        except ValueError:
+            if cls._explicit_custom_unit_regex.match(t.value):
```

Comment:
Moved from the original `_parse_unit` - so that any custom units are defined *after* possible aliases are checked. Arguably perhaps even a bit more logical.